Update toZonedDateTime documentation. (#1149)
tc39/proposal-temporal: docs/date.md

This is because it's not possible to compare `Temporal.PlainDate` objects with the relational operators `<`, `<=`, `>`, or `>=`.
Use `Temporal.PlainDate.compare()` for this, or `date.equals()` for equality.

### date.**toZonedDateTime**(_item_: object) : Temporal.ZonedDateTime

**Parameters:**

> - `item` (object): an object with properties to be added to `time`. The following properties are recognized:

Reviewer: ```suggestion
- `item` (object): an object with properties to be combined with `date`. The following properties are recognized:
```

>   - `time` (optional `Temporal.PlainTime` or value convertible to one): a time of day on `date` used to merge into a `Temporal.ZonedDateTime`.

Reviewer: ```suggestion
  - `time` (optional `Temporal.PlainTime` or value convertible to one like `'10:00'`): a time of day on `date` used to merge into a `Temporal.ZonedDateTime`.
```

>   - `timeZone` (required `Temporal.TimeZone` or value convertible to one, or an object implementing the [time zone protocol](./timezone.md#protocol)): the time zone in which to interpret `date` and `time`.

Reviewer: ```suggestion
  - `timeZone` (required `Temporal.TimeZone`, a value convertible to one like `'Asia/Tokyo'`, or an object implementing the [time zone protocol](./timezone.md#protocol)): the time zone in which to interpret `date` and `time`.
```

Here's another idea for editing this content. It's an easier edit than rewriting the sentence again, and probably more technically accurate than my previous edit in the Instant PR. ;-)
